crates/contract/src/call.rs

```diff
@@ -190,25 +187,25 @@ impl CallDecoder for () {
 /// [sol]: alloy_sol_types::sol
 #[derive(Clone)]
 #[must_use = "call builders do nothing unless you `.call`, `.send`, or `.await` them"]
-pub struct CallBuilder<P, D> {
+pub struct CallBuilder<N: Network, P, D> {
     // TODO: this will not work with `send_transaction` and does not differentiate between EIP-1559
     // and legacy tx
-    request: TransactionRequest,
+    request: N::TransactionRequest,
     block: Option<BlockId>,
     state: Option<StateOverride>,
     provider: P,
     decoder: D,
 }
 
 // See [`ContractInstance`].
-impl<P: TempProvider> DynCallBuilder<P> {
+impl<N: Network, P: Provider<N>> DynCallBuilder<N, P> {
```

Comment:
This will only accept BoxTransport because of `Provider<N, T = BoxTransport>`, I think we need to make this generic over T: Transport as well